iter/itergen.go

```diff
@@ -0,0 +1,37 @@
+package iter
+
+import (
+	"github.com/dgryski/go-tsz"
+	"github.com/raintank/worldping-api/pkg/log"
+)
+
+type IterGen struct {
+	b   []byte
+	ts  uint32
+	len uint64
+}
+
+func NewGen(b []byte, ts uint32, len uint64) IterGen {
+	return IterGen{
+		b,
+		ts,
+		len,
+	}
+}
+
+func (ig *IterGen) Get() (*Iter, error) {
+	it, err := tsz.NewIterator(ig.b)
+	if err != nil {
+		log.Error(3, "failed to unpack cassandra payload. %s", err)
```

Comment:
logging should be left up to the caller. inside lowlevel structures such as IterGen we can't assume any particular logging approach